Source/Painting/SvgPatternServer.cs

```diff
@@ -163,71 +157,84 @@ public override Brush GetBrush(SvgVisualElement renderingElement, ISvgRenderer r
                 curr = SvgDeferredPaintServer.TryGet<SvgPatternServer>(curr.InheritGradient, renderingElement);
             } while (curr != null);
 
-            var childElem = chain.Where(p => p.Children != null && p.Children.Count > 0).FirstOrDefault();
-            if (childElem == null) return null;
-            var widthElem = chain.Where(p => p.Width != null && p.Width != SvgUnit.None).FirstOrDefault();
-            var heightElem = chain.Where(p => p.Height != null && p.Height != SvgUnit.None).FirstOrDefault();
-            if (widthElem == null && heightElem == null) return null;
+            var hasChildren = chain.Where(p => p.Children.Count > 0).FirstOrDefault();
+            if (hasChildren == null)
+                return null;
+            var hasX = chain.Where(p => p.X != null && p.X != SvgUnit.None).FirstOrDefault();
+            var hasY = chain.Where(p => p.Y != null && p.Y != SvgUnit.None).FirstOrDefault();
+            var hasWidth = chain.Where(p => p.Width != null && p.Width != SvgUnit.None).FirstOrDefault();
+            var hasHeight = chain.Where(p => p.Height != null && p.Height != SvgUnit.None).FirstOrDefault();
+            if (hasWidth == null || hasHeight == null)
+                return null;
+            var hasPatternUnit = chain.Where(p => p.PatternUnits != SvgCoordinateUnits.Inherit).FirstOrDefault();
+            var hasPatternContentUnit = chain.Where(p => p.PatternContentUnits != SvgCoordinateUnits.Inherit).FirstOrDefault();
+            var hasViewBox = chain.Where(p => p.ViewBox != null && p.ViewBox != SvgViewBox.Empty).FirstOrDefault();

```

Comment:
Same for all these variable names starting with `has` - they all suggest Boolean values, while they are objects, and are later used as objects, so this reads a bit confusing.